pkg/util/validation/stack_test.go

```diff
@@ -0,0 +1,54 @@
+package validation
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateStackTag(t *testing.T) {
+	t.Run("valid tag", func(t *testing.T) {
+		tags := map[apitype.StackTagName]string{
+			"tag-name": "tag-value",
```

Comment:
Might I suggest some more "classes" of tags here?

What about:

- "-"
- ".."
- "foo:bar.baz"
- "__underscores"
- "AaBb123"